### ISSUE TITLE
Shows the secret information for a brief moment before hiding it.

### ISSUE THREAD

**Issue by olafurw** [1nVitr0/plugin-vscode-censitive#50]
When I open my `.env` file, for a very brief moment the secret is visible.

Making the plugin not usable for it's main purpose.
I'm using Windows 11, WSL 22.04 and I installed the plugin and I'm using the default settings.

(don't worry, there is no actual secret here)

https://github.com/user-attachments/assets/c31a04a9-8a78-41d9-954e-479b7350e93a

**Comment by 1nVitr0** [comment]
To my knowledge there is not much that can be done here. VS Code loads the file and then passes it on to extensions to provide additional syntax support, code actions or in the case of this extension code decorations. As far as I am aware there is nothing that can be done to hide a file's content until the decorators have been loaded in. Aside from making the extension as efficient as possible to minimize the delay.

This was originally intended for streaming, but is more of a coworker protection, so that you don't display your credential to the entire internet cafe.

I will do some research into possible API extension, but I don't expect much changes there. There might be a possibility to get rid of the delay when switching between already open files like in your example above.  If I don't get back to you by the end of next week, fell free to annoy me here.

**Comment by 1nVitr0** [comment]
Yeah, there hasn't been any update to the API. Unfortunately VSCode always detaches the decorations when changing editors. Feel free to upvote https://github.com/microsoft/vscode/issues/136241 for a `persistent` property that can be used by extension authors. Even wih a `persistent` property, there won't be a fix for the delay when initially opening a file, there will always be a gap between VSCode displaying the text and decorations being applied.

I did spot an error in my code though, that may delay censoring in some cases. Won't be noticable in most cases, but can't hurt to fix
